WordPress/gutenberg
# Data Module: Support controls in resolvers

This PR tweaks the data module to allow extended usage of the controls API in resolvers. To achieve that it does:

 - Refactors the way resolvers are registered and fulfilled in the data module
 - It exposes a `fulfill` function in the registry essentially to allow plugins to override the fulfillment implementation.
 - Refactors the redux-routine middleware to allow the possibility to know when the execution of the routine ends
 - Extend the `fulfill` function in the controls plugin to support resolvers in the controls plugin.

**Testing instructions**

 - Check that the current resolvers work properly (we don't use the controls plugin yet in Gutenberg)
